test/hotspot/jtreg/runtime/os/TestExplicitPageAllocation.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package runtime.os;
+
+/* @test
+ * @bug 8271195
+ * @summary Use largest available large page size smaller than LargePageSizeInBytes when available.
+ * @requires os.family == "linux"
+ * @requires vm.gc != "Z"
+ * @library /test/lib
+ * @build jdk.test.whitebox.WhiteBox
+ * @run driver jdk.test.lib.helpers.ClassFileInstaller jdk.test.whitebox.WhiteBox
+ * @run main/othervm -Xbootclasspath/a:. -XX:+UnlockDiagnosticVMOptions -XX:+WhiteBoxAPI runtime.os.TestExplicitPageAllocation
+ */
+
+import jdk.test.lib.process.ProcessTools;
+import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.Utils;
+import java.util.regex.Pattern;
+import java.util.regex.Matcher;
+import java.util.List;
+import java.io.*;
+import java.util.Scanner;
+import jdk.test.whitebox.WhiteBox;
+
+public class TestExplicitPageAllocation {
+
+    private static final String DIR_HUGE_PAGES = "/sys/kernel/mm/hugepages/";
+    private static final long HEAP_SIZE_IN_KB = 1 * 1024 * 1024;
+
+    private static final Pattern HEAP_PATTERN = Pattern.compile("Heap:");
+    private static final Pattern PAGE_SIZE_PATTERN = Pattern.compile(".*page_size=([^ ]+).*");
+    private static final Pattern HUGEPAGE_PATTERN = Pattern.compile(".*hugepages-([^ ]+).*kB");
+
+    private static String errorMessage = "";
+
+    private static final int MAX_NUMBER_OF_PAGESIZE = 64;
+    private static boolean[] pageSizes = new boolean[MAX_NUMBER_OF_PAGESIZE];
+    private static int[] pageCount = new int[MAX_NUMBER_OF_PAGESIZE];
+    private static int vmPageSizeIndex;
+    private static final WhiteBox wb = WhiteBox.getWhiteBox();
+
+    public static void main(String args[]) {
+        try {
+            doSetup();
+            for (int i = MAX_NUMBER_OF_PAGESIZE - 1; i > vmPageSizeIndex; i--) {
+                if (pageSizes[i]) {
+                    testCase(i);
+                    break;
```

Comment:
Can you please motivate this break? Why do we only want to test the largest page size? And if this is the case why not just record the largest page size during the setup-phase?